### ISSUE TITLE
Cloud Run v2 service `cloud_sql_instance` list ordering

### ISSUE THREAD

**Issue by au2001** [hashicorp/terraform-provider-google#13368]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request.
* Please do not leave _+1_ or _me too_ comments, they generate extra noise for issue followers and do not help prioritize the request.
* If you are interested in working on this issue or have submitted a pull request, please leave a comment.
* If an issue is assigned to the `modular-magician` user, it is either in the process of being autogenerated, or is planned to be autogenerated soon. If an issue is assigned to a user, that user is claiming responsibility for the issue. If an issue is assigned to `hashibot`, a community member has claimed the issue already.

### Terraform Version

```sh
$ terraform -v
Terraform v1.3.6
on linux_amd64
+ provider registry.terraform.io/hashicorp/google v4.46.0
+ provider registry.terraform.io/hashicorp/google-beta v4.47.0
```

### Affected Resource(s)

* `google_cloud_run_v2_service`

### Terraform Configuration Files

```tf
resource "google_cloud_run_v2_service" "backend" {
  name     = "backend-${var.environment}"
  location = "europe-west1"

  template {
    containers {
      image = "..."

      volume_mounts {
        mount_path = "/cloudsql"
        name       = "cloudsql"
      }
    }

    volumes {
      name = "cloudsql"

      cloud_sql_instance {
        instances = [
          google_sql_database_instance.a.connection_name,
          google_sql_database_instance.b.connection_name
        ]
      }
    }
  }
}
```

### Debug Output

```tf
$ terraform plan
Terraform used the selected providers to generate the following execution plan.
Resource actions are indicated with the following symbols:
  ~ update in-place

Terraform will perform the following actions:

  # module.staging.google_cloud_run_v2_service.backend will be updated in-place
  ~ resource "google_cloud_run_v2_service" "backend" {
        id                      = "projects/xyz/locations/europe-west1/services/backend-staging"
        name                    = "backend-staging"
        # (19 unchanged attributes hidden)

      ~ template {
            # (7 unchanged attributes hidden)

          ~ volumes {
                name = "cloudsql"

              ~ cloud_sql_instance {
                  ~ instances = [
                      - "xyz:europe-west1:b",
                        "xyz:europe-west1:a",
                      + "xyz:europe-west1:b",
                    ]
                }
            }

            # (3 unchanged blocks hidden)
        }

        # (1 unchanged block hidden)
    }

Plan: 0 to add, 1 to change, 0 to destroy.

────────────────────────────────────────────────────────────────────────────────

Note: You didn't use the -out option to save this plan, so Terraform can't
guarantee to take exactly these actions if you run "terraform apply" now.
```

### Expected Behavior

The `instances` array in the `cloud_sql_instance` block should not cause a change because of the reordering of instances by the Google Cloud API.\
Or the order of resources should be predictable based on their name.\
Currently, the order never changes for a given set of Cloud SQL instances and a given Cloud Run service, but creating resources with similar names (ex. in a different project, or by changing a suffix) will cause a different order.

### References

* [hashicorp/terraform-provider-aws#22274](https://github.com/hashicorp/terraform-provider-aws/issues/22274) - Same issue in the AWS provider on the `aws_iam_policy_document` resource
* b/272363402

**Comment by edwardmedia** [comment]
Server side reordering. Need to change the [type](https://github.com/hashicorp/terraform-provider-google/blob/main/google/resource_cloud_run_v2_service.go#L466) from list to set. A breaking-change.

**Comment by ScottSuarez** [comment]
Implemented this change for the 5.0 release
https://github.com/GoogleCloudPlatform/magic-modules/pull/8929
